### DISCUSSION TITLE
Variable batch size for ranking problems

### DISCUSSION THREAD

**Discussion by AlessandroGussoni** [pyg-team/pytorch_geometric#8108] [Q&A]
Hi everyone! I am relatively new to GNNs and looking for guidance.
I am working on a problem with n different graphs with node features for each of them. For each graph i have a variable number of graph features, for example for graph 1 i am going to have 100 different graph features, for graph 2 150 and so on. My task is a sorting problem at graph level; each of the graph vector of features will have an associated label, my goal is to sort the 100 labels related to graph 1 to lowest to highest. What i would like to do is to put each graph in a different "batch", so i can use a ranking loss on the full list of prediction for each graph, is there any available resources that show how to do it? Hope that what i wrote makes sense!

**Comment by anthonysirico** [comment]
If i am reading this correctly, you want to separate your graphs by different criteria into different datasets.  May  I  suggest:

```
def process(self):
    dataset1 = []
    dataset2 = []
    ...

    for index, graph in df.iterrows():  # assuming its a dataframe
        ...

        if graphfeatures == ???:
            data = Data(x=x, y=y...)
            dataset1.append(data)
....
```
I am sure someone else has a better idea, but that's my thought.
